spec/controllers/trainees/training_routes_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe Trainees::TrainingRoutesController do
+  let(:user) { build(:user, provider: trainee.provider) }
+  let(:trainee_params) { { training_route: :provider_led_postgrad } }
+
+  before do
+    allow(controller).to receive(:current_user).and_return(user)
+  end
+
+  describe "#update" do
+    subject { post(:update, params: { trainee_id: trainee.slug, trainee: trainee_params }) }
+
+    context "when changing to the same route" do
+      context "with a completed course details section" do
+        let(:trainee) { create(:trainee, :provider_led_postgrad, :with_course_details) }
+
+        it "does not clear the course details section of the trainee" do
+          subject
+          trainee.reload
+          expect(trainee.progress.course_details).to eq nil
+          expect(trainee.course_code).to eq trainee.course_code
```

Comment:
`be_present?` may be appropriate as it'll explicitly check the value isn't `nil` otherwise (hypothetically) you could be comparing `nil` with `nil` and this test would pass.

Reply:
All done 👍